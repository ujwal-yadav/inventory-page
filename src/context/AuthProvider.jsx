import { useContext, createContext } from 'react';
import { useAuthProvider } from './AuthProvider.hooks';
const AuthContext = createContext();

const AuthProvider = ({ children }) => {
  const { token, login, logOut } = useAuthProvider();

  return (
    <AuthContext.Provider value={{ token, login, logOut }}>
      {children}
    </AuthContext.Provider>
  );
};

export default AuthProvider;

export const useAuth = () => {
  return useContext(AuthContext);
};
