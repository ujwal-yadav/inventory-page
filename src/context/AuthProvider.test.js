import React from 'react';
import { render, screen } from '@testing-library/react';
import AuthProvider from './AuthProvider';

jest.mock('./AuthProvider.hooks', () => ({
  useAuthProvider: jest.fn(),
}));

describe('AuthProvider', () => {
  it('should render correctly', () => {
    const mockToken = 'mock-token';
    const mockLogin = jest.fn();
    const mockLogOut = jest.fn();

    require('./AuthProvider.hooks').useAuthProvider.mockReturnValue({
      token: mockToken,
      login: mockLogin,
      logOut: mockLogOut,
    });

    const { asFragment } = render(<AuthProvider>{'UI'}</AuthProvider>);

    expect(asFragment()).toMatchSnapshot();
  });
});
