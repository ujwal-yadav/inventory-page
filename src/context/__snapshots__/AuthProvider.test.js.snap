// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthProvider should render correctly 1`] = `
<DocumentFragment>
  UI
</DocumentFragment>
`;
